### ISSUE TITLE
Add default arg "smart insert" configurability.

### ISSUE THREAD

**Issue by pq** [flutter/flutter-intellij#831]
See: https://github.com/JetBrains/intellij-plugins/pull/495#issuecomment-287013985

Like `AngularJSConfigurationPage`, `WebEditorOptionsProvider` and other usages of `EditorSmartKeysConfigurableEP`, we should surface a way to opt-out of our new "smart insert"s of default args.

![0bc137d2-0a4a-11e7-9637-9abbad49659f](https://cloud.githubusercontent.com/assets/67586/24005094/4960ea1c-0a25-11e7-85df-16d56b533781.png)

cc @alexander-doroshko

**Comment by pq** [comment]
@alexander-doroshko: I started implementing this in Smart Keys and then noticed that we already have a tab for codegen options.

<img width="520" alt="screen shot 2017-03-16 at 9 33 07 am" src="https://cloud.githubusercontent.com/assets/67586/24007376/9ad8e5e2-0a2b-11e7-8c3c-6b54b89716ff.png">

Any reason we shouldn't put it here?

(It'd be nice to have one less place folks need to know to look for Dart-relevant settings.)

**Comment by alexander-doroshko** [comment]
This Code Generation tab on you screenshot is inside the Code Style section. Live templates in code completion do not look related to Code Style. No other (and we have many of them) Gode Generation tabs has anything about code completion.

I'd expect such an option around Editor | General | Code Completion. 

But I noticed that other similar things are in the Smart Keys page. For example in IDEA:
![image](https://cloud.githubusercontent.com/assets/4499789/24008079/9bd9c352-0a81-11e7-964a-bcda83480236.png)

Or the best example comes from PhpStorm: it [already has an option](https://blog.jetbrains.com/phpstorm/2016/07/parameter-code-completion-in-phpstorm-2016-2/) for args template, though not sure the option text is good:
![image](https://cloud.githubusercontent.com/assets/4499789/24008145/d462b51c-0a81-11e7-9484-8492884bf792.png)

**Comment by pq** [comment]
OK.  Not disagreeing with how things are organized -- from an implementation perspective these things make perfect sense.  It would just be a bit nicer if settings weren't spread all over the place IMO.  I doubt if users know (or care to know) about some of these distinctions.  (I certainly didn't! 😉 )

Regardless, it doesn't look like the codegen tab is where this should go so I'll throw together a Smart Keys entry.  Down the road if there's any consolidation to be done we can revisit.

Thanks for the quick feedback!

**Comment by pq** [comment]
Should have a PR out by end of day but here's the WIP in case anyone cares to word-smith.

<img width="429" alt="screen shot 2017-03-16 at 11 37 26 am" src="https://cloud.githubusercontent.com/assets/67586/24012921/f8daa6ba-0a3c-11e7-81ad-529239f305b4.png">

@devoncarew @mit-mit @bwilkerson?

**Comment by bwilkerson** [comment]
Do we need an option? Do we want to allow this behavior to be disabled? It seems generally useful to me.

**Comment by pq** [comment]
I tend to think it will be generally useful but @alexander-doroshko was asking us to allow for disablement.  I hope to default it to ON.

**Comment by alexander-doroshko** [comment]
There are so many users with different expectations... Yesterday I was asked not to insert closing paren when completing method call :). A user who disabled 'Insert paired bracket' wants to end up with `methodName(|` after selecting an item from completion.

We experimented with similar feature in Java internally for several months but so far even didn't make the option visible (and it is off by default).

Well, if we target the feature to 2017.2 then I'm ok to live for a couple of months with EAPs without an option to collect feedback.

**Comment by pq** [comment]
Thanks for the background, @alexander-doroshko!

Here's a PR to get us started: https://github.com/JetBrains/intellij-plugins/pull/497

I'll leave it to you to decide whether this should target `2017.1` or wait on `2017.2` 👍

**Comment by alexander-doroshko** [comment]
I just thought you want me to release it earlier. If 2017.2 is ok then we can start EAPs without an option. If you want to see it in 2017.1 then having an option sounds safer.

**Comment by pq** [comment]
We'd love to see the feature landed ASAP so whatever works.  I'm comfortable with or without the option cherry-picked in though I'd probably prefer it included.  Your call.

**Comment by pq** [comment]
Landed in https://github.com/JetBrains/intellij-plugins/commit/7697d0f9473294424c77be7d836a1219eca07865.
